### DISCUSSION TITLE
Way of providing an external values.yaml file

### DISCUSSION THREAD

**Discussion by raelix** [argoproj/argo-cd#9480] [Q&A]
Hi everyone,
I'm using argocd and it works really well. What I'm trying to achieve is to provide different values.yaml files based on prod/dev. The problem I'm facing is that almost the chart I'm installing are third-party charts hence I can't add override values in their projects.
I also would like to avoid passing the values inside the application definition due to the large amount of parameters that I have to pass, moreover using an umbrella chart is not the best approach because all the values must be under a different root section.
Is there a way to pass them in a standard way? At the end, helm supports external values.yaml override

Thanks in advance

**Comment by crenshaw-dev** [comment]
That feature is currently expected in 2.5. https://github.com/argoproj/argo-cd/issues/2789

**Reply by raelix** [reply]
Thanks for the reply. When the 2.5 should be released?

Thanks in advance

**Reply by VladoPortos** [reply]
Looking forward to it, its kind of needed since people might use official helm repo, but want to use own values.yaml hosted elsewhere.

**Reply by crenshaw-dev** [reply]
We expect to release 2.5 by the end of August.

**Reply by pojusoi** [reply]
In our project we are also very interested from this feature.

**Most important is,** that information from a deployment cannot be shared to product line. This is because we are handling information that are under privacy regulations in those countries where we are operating. We must be able to separate the values from the deployment (from the product).

In our case we are running 5 different data centers (world-wide) and we are replicating equal amount of deployments from one product to different regions. Each regions are different, in many means but most challenges we have comes from a privacy regulations.


So how does the planned schedule look like... I see version 2.5.0-rc1 available bur in a release notes I cannot find anything related.

Can you share more details from a feature and plan for a delivery?

**Reply by crenshaw-dev** [reply]
@pojusoi we had to push this feature to 2.6. There is an open PR, but it just wasn't ready in time for 2.5. You can track the issue here: https://github.com/argoproj/argo-cd/issues/2789

Yours is a really interesting use case. I would caution that, if the information in the values.yaml file is _secret_ (in the way that an encryption key or a password is secret), then the external-values-file feature is not an appropriate solution. The Argo CD repo-server component is not designed to handle secret information.

If the information in the values.yaml file is merely _sensitive_, then the new feature should be suitable.

You can read more about the feature here: https://github.com/argoproj/argo-cd/pull/10432

The PR includes docs updates with examples.

**Reply by pojusoi** [reply]
@crenshaw-dev thank you for details, ref: https://github.com/argoproj/argo-cd/pull/10432

This feature will solve our problems and is very much welcome to our toolbox.

**Reply by ido-vcita** [reply]
@crenshaw-dev Do we have an update regarding the release date?

**Reply by dorzvulunvcita** [reply]
This feature is super important to our implementation of ArgoCD. For my company it will be a game changer! hope you'll push this asap.

**Reply by crenshaw-dev** [reply]
@ido-vcita the expected release date for the first 2.6 release candidate is the end of November. Since we pushed 2.5 back quite a bit, I'm hoping we can be a lot more aggressive about sticking with the November release date.

**Comment by reddy2018** [comment]
is there any way to pass the values.yaml file inside the argocd dashboard while creating the application?

**Reply by reddy2018** [reply]
i am trying to create metallb but need to pass one values.yaml file to override the values. 

can you please share me if you have any solution for that
